lib/puppet/reports/servicenow.rb

```diff
@@ -8,6 +8,57 @@
   def process
     settings_hash = settings
 
+    case settings['operation_mode']
+    when 'event_management'
+      process_event_management(settings_hash)
+    else
+      # default to incident management
+      process_incident_management(settings_hash)
+    end
+  rescue StandardError => e
+    Puppet.err "servicenow report processor error: #{e}\n#{e.backtrace}"
+  end
+
+  def process_event_management(settings_hash)
+    event_data = {
+      'source'   => 'Puppet',
+      'type'     => 'node_report',
+      # 5 => 'OK' severity
+      'severity' => '5',
+    }
+
+    # Compute the message key hash, which contains all relevant information
+    # involved in determining the event's uniqueness. In our case, this
+    # information's the node and report.
+    message_key_hash = {
+      'node'   => host,
+      'report' => calculate_report_message_key_hash(status, resource_statuses),
+    }
+
+    # Finally calculate the message key. This is the SHA-1 of the message key
+    # hash's raw JSON encoding (sorted).
+    event_data['message_key'] = Digest::SHA1.hexdigest(message_key_hash.to_json.chars.sort.join)
```

Comment:
This felt simpler to do than a SHA-1 hash of the message hash's raw JSON encoding with sorted keys (where sorted keys would require some deepsort logic which I don't think is necessary here).